targets/export-elasticsearch/src/repositories/s3.ts

```diff
@@ -0,0 +1,187 @@
+    const data: string = response.data;
+    const command = new PutObjectCommand({
+      Bucket: this.bucketName,
+      Key: keyFolder,
+      Body: data,
+      ContentType: "text/xml",
+      ACL: "public-read",
+    });
+    await this.s3Client.send(command);
+  }
+
+  async uploadAgreements(
+    environment: Environment,
+    data: string,
+    folder: string,
+    key: string
+  ): Promise<void> {
+    const keyFolder =
+      environment === "production"
+        ? `${this.publishedFolder}/${folder}/${key}`
+        : `${this.previewFolder}/${folder}/${key}`;
+    const command = new PutObjectCommand({
+      Bucket: this.bucketName,
+      Key: keyFolder,
+      Body: data,
+      ContentType: "application/json",
+      ACL: "public-read",
+    });
+    await this.s3Client.send(command);
+  }
+
+  async copyFolder(environment: Environment): Promise<void> {
+    try {
+      const copyFolder =
+        environment === "production"
+          ? `${this.previewFolder}/${this.defaultFolder}`
+          : `${this.draftFolder}/${this.defaultFolder}`;
+      const pasteFolder =
+        environment === "production"
+          ? `${this.publishedFolder}/${this.defaultFolder}`
+          : `${this.previewFolder}/${this.defaultFolder}`;
+
+      // 1. Récupérer les anciennes clés dans le published
+      const listCommandPasteFolder = new ListObjectsCommand({
+        Bucket: this.bucketName,
+        Prefix: pasteFolder,
+      });
+      const listContentsPasteFolder = await this.s3Client.send(
+        listCommandPasteFolder
+      );
+      const listKeysPasteFolder =
+        listContentsPasteFolder.Contents?.map((file) => {
+          if (!file.Key) {
+            throw new Error("File key is not defined");
+          }
+          return file.Key;
+        }) ?? [];
+
+      // 2. Récupérer les nouvelles clés dans la preview
+      const listCommandCopyFolder = new ListObjectsCommand({
+        Bucket: this.bucketName,
+        Prefix: copyFolder,
+      });
+      const listContentsCopyFolder = await this.s3Client.send(
+        listCommandCopyFolder
+      );
+      const listKeysCopyFolder =
+        listContentsCopyFolder.Contents?.map((file) => {
+          if (!file.Key) {
+            throw new Error("File key is not defined");
+          }
+          return file.Key;
+        }) ?? [];
+
+      // 3. Copier les clés de la preview vers le published
+      for (const key of listKeysCopyFolder) {
+        const nameFile = key.split("/").pop();
+        const mimeType = mime.lookup(key);
+        const copyCommand = new CopyObjectCommand({
+          Bucket: this.bucketName,
+          CopySource: encodeURI(`${this.bucketName}/${key}`),
+          Key: `${pasteFolder}/${nameFile}`,
+          ACL: "public-read",
+          MetadataDirective: "REPLACE",
+          ContentType: mimeType,
+        });
+
+        await this.s3Client.send(copyCommand);
+        await new Promise((resolve) => setTimeout(resolve, 200));
```

Comment:
Pourquoi y a t il un setTimeout de 200 ici ?

Reply:
Pour éviter d'avoir plusieurs call qui surchagent l'API S3, on peut l"enlever, mais bon, jamais trop fan de faire pleins de calls dans un `for`

Reply:
Je l'ai retiré